libs/unity/library/Runtime/Scripts/Media/MediaLine.cs

```diff
@@ -248,10 +276,13 @@ internal MediaLine(PeerConnection peer, MediaKind kind)
             _mediaKind = kind;
         }
 
-        protected void UpdateTransceiverDesiredDirection()
+        private void UpdateTransceiverDesiredDirection()
         {
             if (Transceiver != null)
             {
+                // Avoid races on the desired direction by limiting changes to the main thread.
+                _peer.EnsureIsMainAppThread();
```

Comment:
Move out if `if()`? Having a precondition on the function that depends on some other property is weird. Do we really need to call that method with `Transceiver == null` from another thread than the main Unity app thread?

Reply:
`EnsureIsMainAppThread` will throw if the peer is not awake, even if the thread is correct. So we are checking only when we know that it is awake (if it isn't, the method will be a nop so no one cares). Added a comment.